### PR TITLE
Update policy status filter queries.

### PR DESCRIPTION
**Pull request by terloon**
Using the diagram below as reference to help identify policy status.

Screen Shot 2021-09-17 at 11 20 51 AM

## 📚 Purpose
*Fix getting back incorrect policies for given filters.*

## 👌 Resolves:
- [ ] ✨ APPS-93 Helps with the implementation of this feature.

## 📦 Impacts:
*mds-policy-service*

## ✅ PR Checklist
- [ ] Check that only certain policies are returned for given filters.

### CODE REVIEW

File: packages/mds-policy-service/repository/index.ts [hunk 1]
```diff
@@ -80,24 +84,35 @@ class PolicyReadWriteRepository extends ReadWriteRepository {
...
           [key in Exclude<POLICY_STATUS, 'unknown'>]: { expression: string; params?: object }
         } = {
-          draft: { expression: "policy_json->>'publish_date' IS NULL" },
-          deactivated: { expression: 'superseded_by IS NOT NULL AND array_length(superseded_by, 1) >= 1' },
+          draft: {
+            expression: [NOT_PUBLISHED].join(DELIMITER)
+          },
+          deactivated: {
+            expression: [SUPERSEDED].join(DELIMITER)
+          },
```
Comment (avatarneil) [inline packages/mds-policy-service/repository/index.ts:104]: I could forsee a conflict where `SUPERSEDED` and `NOT_PUBLISHED` are both true for some things (if you write a new policy referencing an unpublished policy, if that's even possible, which maybe it shouldn't be)

**Review by avatarneil — APPROVED** [review]
LGTM, one comment

**Review by nessur — APPROVED** [review]

File: packages/mds-policy-service/repository/index.ts (hunk 1 above)
Comment (terloon) [reply packages/mds-policy-service/repository/index.ts:104]: True. Who would be the source of truth on the policy status definitions? They can clarify whether "deactivated" can only be for policies that have been published. Also whether to add a check to the "supersede" action to only work for policies not in draft state (published).

**Review by BigIdeas — APPROVED** [review]

File: packages/mds-policy-service/repository/index.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-policy-service/repository/index.ts:104]: Probably @nelsonsantryl I'd imagine

Comment (nelsonsantryl) [reply packages/mds-policy-service/repository/index.ts:104]: Hopefully I can clarify:

- A "deactivated" policy, must have been active at some point previously--> therefore the policy was published.
- The workflow to create a policy that supersedes another, is to go to the original **published** policy, and "clone" it (which will create the link back to the OG policy).  So yes you are correct.  There is no way to supersede a policy that is in a "draft" status.

Let me know if anything needs further clarification.

**Pull request merged**